### PR TITLE
Fix for Issue 716: Moving example from cluster folder to top level folder

### LINKED ISSUE #716: Move examples folder from cluster to the root directory
### What problem are you facing?

The current examples folder is buried in [cluster] folder, which has two issues:
- Lower visibility for users who want to try the examples
- Lack of logical organization, as the cluster folder is mostly addressing other areas

[cluster]: https://github.com/crossplaneio/crossplane/tree/master/cluster

### PR DESCRIPTION
**Pull request by nchetan**
### Description of your changes
Move examples folder under cluster folder to top level.
Changes are verified with following commands
* prior to changes make, make test, make e2e, make reviewable, make (to ensure nothing is broken before changes)
* make the move
* Verify move with make, make test, make e2e, make reviewable, make

No issues are discovered with these commands.

Fixes #716 

I have:

- [ X] Read and followed Crossplane's [contribution process].
- [ X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

As mentioned above, with commands
make test
make e2d
make reviewable
make
[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by hasheddan — CHANGES_REQUESTED** [review]
@nchetan thanks so much for tackling this! Would you mind making updates to links that will be broken by this move? You can use [this query](https://github.com/crossplane/crossplane/search?q=%22cluster%2Fexamples%22&unscoped_q=%22cluster%2Fexamples%22) to find the applicable links. You can omit the links in the [release process doc](https://github.com/crossplane/crossplane/blob/d69066ddb3ca1ded38a9dc43adb82e3adb8c1dcf/docs/release/release-process.md) as it references a specific branch :+1:

File: docs/release/release-process.md [hunk 1]
```diff
@@ -185,7 +185,7 @@ examples to point to the new versions that we will be releasing soon.
   guides.
   * searching for `:master` will help a lot here
 * Examples, such as [`StackInstall` yaml
-  files](https://github.com/crossplane/crossplane/tree/release-0.9/cluster/examples/provider)
+  files](https://github.com/crossplane/crossplane/tree/release-0.9/examples/provider)
```
Comment (hasheddan) [inline docs/release/release-process.md:188]: Could you omit this update as the existing path is still correct for the release-0.9 branch? 🙂

Comment (nchetan) [reply docs/release/release-process.md:188]: @hasheddan Will do. Do you want me to amend the commit and force push it OR make another commit?
Not sure what is the git workflow your org follows. Hence the Q.

Comment (hasheddan) [reply docs/release/release-process.md:188]: Force push is good 👍

Comment (nchetan) [reply docs/release/release-process.md]: @hasheddan The changes have been force pushed to https://github.com/nchetan/crossplane. Do I have to re-request?

**Review by hasheddan — APPROVED** [review]
This looks great @nchetan! Thanks for your work here and congrats on your first contributions to Crossplane!

**Pull request merged**